analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TestClassShouldHaveTestMethod.MsTest.cs

```diff
@@ -113,6 +113,72 @@ public static void AfterTests()
         {
         }
     }
+
+    [TestClass]
+    public class SetupAttributes4 // Noncompliant
+    {
+        [TestInitialize]
+        public void BeforeTests(TestContext context)
+        {
+        }
+    }
+
+    [TestClass]
+    public class SetupAttributes5 // Noncompliant
+    {
+        [TestCleanup]
+        public void AfterTests()
+        {
+        }
+    }
+
+
+    [TestClass]
+    public class SetupAttributes6 // Noncompliant
+    {
+        [TestInitialize]
+        public void BeforeTests(TestContext context)
+        {
+        }
+
+        [TestCleanup]
+        public void AfterTests()
+        {
+        }
+    }
+
+    [TestClass]
+    public class SetupAttributes7 // Noncompliant
+    {
+        [ClassInitializeAttribute]
+        public void BeforeTests(TestContext context)
+        {
+        }
+    }
+
+    [TestClass]
+    public class SetupAttributes8 // Noncompliant
+    {
+        [ClassCleanupAttribute]
+        public void AfterTests()
+        {
+        }
+    }
+
+
+    [TestClass]
+    public class SetupAttributes9 // Noncompliant
+    {
+        [ClassInitializeAttribute]
+        public void BeforeTests(TestContext context)
+        {
+        }
+
+        [ClassCleanupAttribute]
+        public void AfterTests()
+        {
+        }
+    }
```

Comment:
Please add the test case with `TestInitialize` and `TestCleanup` in an abstract class. Inherited by a compliant class - can be added to `MyCommonCode1` above.

Reply:
Please add the test case with `TestInitialize` and `TestCleanup` in a class without `TestClass` to make sure we don't raise. To catch these situations in the future